synapse/handlers/appservice.py

```diff
@@ -406,7 +406,11 @@ def _merge_instances(infos: List[JsonDict]) -> JsonDict:
 
             return combined
 
-        return {p: _merge_instances(protocols[p]) for p in protocols.keys()}
+        return {
+            p: _merge_instances(protocols[p])
+            for p in protocols.keys()
+            if len(protocols[p]) > 0
```

Comment:
This can probably just be:

```suggestion
            if protocols[p]
```

But something else seems bogus here -- `_merge_instances` will return an empty dict if there's no protocols passed in. Should that throw an exception or something if we're explicitly filtering those out?

Reply:
I don't think we want the request to fail so throwing seems off. Returning None will return `{"my-protocol": None}` which isn't the same as `{}`, which is what the spec expects us to return if the protocol is incomplete (i.e. it has no appservices implementing it).

Removing the `len` seems sane however.

Reply:
Right, my point was mostly that it was redundant Removing it (like you did) seems reasonable.

Reply:
Ah, I see. Good good.